### PR TITLE
Add Triangle exercise

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: exercises/triangle/examples/src/Triangle.purs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Data.Either (Either(Left, Right))
+import Data.Generic (class Generic, gShow, gEq)
+
+
+data Triangle
+  = Equilateral
+  | Isosceles
+  | Scalene
+
+derive instance genericTriangle :: Generic Triangle
```
Comment (lpil) [inline exercises/triangle/examples/src/Triangle.purs]: I don't understand this statement and I couldn't find any info on it in the Purescript book. Could you point me in the direction of an explanation of `derive instance`? :)

Comment (paf31) [inline exercises/triangle/examples/src/Triangle.purs]: Yes, there is an explanation in my [24 days of PureScript](https://github.com/paf31/24-days-of-purescript-2016) blog post series (see the Generic deriving post, or Eq and Ord deriving). Also if you only need `eq`, don't use `gEq`, use the `derive` keyword to derive a more efficient `Eq` instance.

File: exercises/triangle/examples/src/Triangle.purs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  eq = gEq
+
+instance showTriangle :: Show Triangle where
+  show = gShow
+
+
+type ErrorMsg = String
+
+
+triangleKind :: Int -> Int -> Int -> Either Triangle ErrorMsg
```
Comment (paf31) [inline exercises/triangle/examples/src/Triangle.purs]: It's a convention to put the error type on the left. That way, all of the instances (Applicative, Monad, etc.) do the expected things.

File: exercises/triangle/examples/src/Triangle.purs (hunk 1 above)
Comment (lpil) [inline exercises/triangle/examples/src/Triangle.purs]: Thank you.

File: exercises/triangle/examples/src/Triangle.purs (hunk 2 above)
Comment (lpil) [inline exercises/triangle/examples/src/Triangle.purs]: Thanks! In Rust there's `Ok` and `Err`, which I find easier to read than `Left` and `Right`. Is there anything like this?

**Comment by lpil** [comment]
Made some changes post feedback :)

File: exercises/triangle/bower.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "ignore": [
+    "**/.*",
+    "node_modules",
+    "bower_components",
+    "output"
+  ],
+  "dependencies": {
+    "purescript-prelude": "^2.1.0",
+    "purescript-console": "^2.0.0",
+    "purescript-either": "~2.0.0",
```
Comment (paf31) [inline exercises/triangle/bower.json]: We should probably use `^` dependencies everywhere.

**Comment by paf31** [comment]
👍 Looks good, thanks!

File: exercises/triangle/bower.json (hunk 3 above)
Comment (lpil) [inline exercises/triangle/bower.json]: I'm not familiar with bower. I added these with `bower install --save purescript-whatever`. Should I do then and then edit the `bower.json` file, or is there a way I can get bower to use `^`?

Comment (paf31) [inline exercises/triangle/bower.json]: I would just edit by hand.

Comment (lpil) [inline exercises/triangle/bower.json]: Done :)

**Comment by paf31** [comment]
Thanks again!

**Pull request merged**